internal/controller/grafana_organization_reconciler.go

```diff
@@ -0,0 +1,96 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+
+	grafanaorganization "github.com/giantswarm/observability-operator/api/v1alpha1"
+	"github.com/pkg/errors"
+	apimachineryerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	"github.com/giantswarm/logging-operator/pkg/logged-cluster/capicluster"
+	loggingconfig "github.com/giantswarm/logging-operator/pkg/resource/logging-config"
+)
+
+// GrafanaOrganizationReconciler reconciles grafanaOrganization CRs
+type GrafanaOrganizationReconciler struct {
+	client.Client
+	Scheme                  *runtime.Scheme
+	LoggingConfigReconciler loggingconfig.Reconciler
+}
+
+//+kubebuilder:rbac:groups=observability.giantswarm.io,resources=grafanaorganizations,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=observability.giantswarm.io,resources=grafanaorganizations/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// It is triggered whenever a change happen to a grafanaOrganization CR and
+// calls the logging config reconciler for each cluster so that their tenant
+// list is always up to date.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.4/pkg/reconcile
+func (g *GrafanaOrganizationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, err error) {
+	logger := log.FromContext(ctx)
+
+	logger.Info("Started reconciling Grafana Organization")
+	defer logger.Info("Finished reconciling Grafana Organization")
+
+	grafanaOrganization := &grafanaorganization.GrafanaOrganization{}
+	err = g.Client.Get(ctx, req.NamespacedName, grafanaOrganization)
+	if client.IgnoreNotFound(err) != nil { // Only handle errors that are not notFound ones
+		return ctrl.Result{}, errors.WithStack(err)
+	}
+
+	clusters := &capi.ClusterList{}
+	err = g.Client.List(ctx, clusters)
+	if err != nil {
+		if apimachineryerrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, errors.WithStack(err)
+	}
+
+	for _, cluster := range clusters.Items {
+		loggedCluster := &capicluster.Object{
+			Object:  &cluster,
+			Options: loggedcluster.O,
+		}
+
+		// Reconcile logging config for each cluster
+		result, err := g.LoggingConfigReconciler.ReconcileCreate(ctx, loggedCluster)
```

Comment:
We should not call `ReconcileCreate` on cluster objects being deleted or when logging is disabled, please use `common.IsLoggingEnabled`, see https://github.com/giantswarm/logging-operator/blob/01a805e8397eca3bf44b04cdcbf0ab1a5428b399/pkg/logging-reconciler/logging_reconciler.go#L28-L32

Reply:
Done :heavy_check_mark: